### PR TITLE
Vessel history/ve 43

### PR DESCRIPTION
**Pull request by fertandil87**
https://globalfishingwatch.atlassian.net/browse/VE-43

translations in another PR because is not clear yet how to build the strings of the events

### CODE REVIEW

File: applications/vessel-history/src/features/profile/components/activity/Activity.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+const Activity: React.FC<InfoProps> = (props): React.ReactElement => {
+  const { t } = useTranslation()
+  const vesselId = useSelector(selectVesselId)
+
+ 
+  const dispatch = useDispatch()
+  useEffect(() => {
+    if (vesselId) {
+      console.log(vesselId)
```
Comment (rdgfuentes) [inline applications/vessel-history/src/features/profile/components/activity/Activity.tsx:30]: 🧹

File: applications/vessel-history/src/features/profile/components/activity/Activity.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+      dispatch(fetchVesselActivityThunk({vesselId}))
+    }
+  }, [dispatch, vesselId])
+
+  const onOpenGroup = useCallback((index: number) => {
+    dispatch(toggleGroup({index}))
+  },[dispatch])
+
+  const eventGroups = useSelector(selectVesselActivity)
+  console.log(eventGroups)
```
Comment (rdgfuentes) [inline applications/vessel-history/src/features/profile/components/activity/Activity.tsx:40]: 🧹

File: applications/vessel-history/src/features/profile/components/activity/Activity.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+            {group.open && group.entries && group.entries.map((event: ActivityEvent, eventIndex) => ( 
+              <Fragment key={eventIndex}>
+                  <div className={styles.event} >
+                    <div  className={styles.eventIcon}>
+                      <i></i>
+                    </div>
+                    <div className={styles.eventData}>
+                      <ActivityDate event={event}/>
+                      <div className={styles.description}>
+                        Fishing in ???? {event.start}
```
Comment (rdgfuentes) [inline applications/vessel-history/src/features/profile/components/activity/Activity.tsx:81]: Missing translation here. What are the `????` ? Is any data that you need to display there missing?

File: applications/vessel-history/src/features/vessels/activity/vessels-activity.slice.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        },
+    },
+    extraReducers: (builder) => {
+        builder.addCase(fetchVesselActivityThunk.pending, (state, action) => {
+            state.status = AsyncReducerStatus.Loading
+
+        })
+        builder.addCase(fetchVesselActivityThunk.fulfilled, (state, action) => {
+            state.status = AsyncReducerStatus.Finished
+            console.log(action.payload)
```
Comment (rdgfuentes) [inline applications/vessel-history/src/features/vessels/activity/vessels-activity.slice.ts:33]: 🧹

File: applications/vessel-history/src/features/vessels/activity/vessels-activity.thunk.ts [hunk 5]
```diff
@@ -0,0 +1,74 @@
+import { createAsyncThunk } from '@reduxjs/toolkit'
+import GFWAPI from '@globalfishingwatch/api-client'
+import { FISHING_EVENTS_DATASET } from 'data/constants'
+import { ActivityEvent, ActivityEventGroup } from 'types/activity'
+
+
+const fetchData = (dataset: string, vesselId: string, signal?: AbortSignal | null) => {
+    console.log(7)
+    return GFWAPI.fetch<any>(
+        `/v1/events?datasets=${encodeURIComponent(dataset)}&vessels=${vesselId}&startDate=2012-01-01&endDate=2021-05-05`
```
Comment (rdgfuentes) [inline applications/vessel-history/src/features/vessels/activity/vessels-activity.thunk.ts:10]: The arguments `startDate` and `endDate` should be received as arguments.

File: applications/vessel-history/src/features/vessels/activity/vessels-activity.thunk.ts [hunk 6]
```diff
@@ -0,0 +1,74 @@
+import { createAsyncThunk } from '@reduxjs/toolkit'
+import GFWAPI from '@globalfishingwatch/api-client'
+import { FISHING_EVENTS_DATASET } from 'data/constants'
+import { ActivityEvent, ActivityEventGroup } from 'types/activity'
+
+
+const fetchData = (dataset: string, vesselId: string, signal?: AbortSignal | null) => {
+    console.log(7)
```
Comment (rdgfuentes) [inline applications/vessel-history/src/features/vessels/activity/vessels-activity.thunk.ts:8]: 🧹

File: applications/vessel-history/src/features/profile/components/activity/Activity.tsx (hunk 3 above)
Comment (fertandil87) [reply applications/vessel-history/src/features/profile/components/activity/Activity.tsx:81]: yes, is not clear what is the data to display from the api

**Pull request merged**